File: app/src/main/java/com/cr4sh/nhlauncher/NHLManager.java
package com.cr4sh.nhlauncher;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class NHLManager {
    private MainActivity mainActivity;
    private ExecutorService executorService;

    private NHLManager() {
        // Private constructor to prevent instantiation
        executorService = Executors.newSingleThreadExecutor();
    }

    private static class Holder {
        private static final NHLManager INSTANCE = new NHLManager();
    }

    public static NHLManager getInstance() {
        return Holder.INSTANCE;
    }

    public MainActivity getMainActivity() {
        return mainActivity;
    }

    public void setMainActivity(MainActivity activity) {
        this.mainActivity = activity;
    }

    public ExecutorService getExecutorService() {
        if (executorService.isShutdown()) {
            executorService = Executors.newSingleThreadExecutor();
        }
        return executorService;
    }

    public void shutdownExecutorService() {
        if (executorService != null && !executorService.isShutdown()) {
            executorService.shutdown();
        }
    }
}

File: app/src/main/java/com/cr4sh/nhlauncher/utils/NHLManager.kt
package com.cr4sh.nhlauncher.utils

import com.cr4sh.nhlauncher.activities.MainActivity

class NHLManager private constructor() {
    private var mainActivity: MainActivity? = null

    private object Holder {
        val INSTANCE = NHLManager()
    }

    companion object {
        @JvmStatic
        fun getInstance(): NHLManager {
            return Holder.INSTANCE
        }
    }

    fun getMainActivity(): MainActivity? {
        return mainActivity
    }

    fun setMainActivity(activity: MainActivity) {
        this.mainActivity = activity
    }
}
